src/models/s2s.h

```diff
@@ -194,29 +196,56 @@ class DecoderS2S : public DecoderBase {
              && opt<std::string>("original-type") == "nematus")  //
         ("skip", opt<bool>("skip"));
 
+    ABORT_IF(opt<bool>("dec-attention-bilinear-lookup"),
+             "--dec-attention-bilinear-lookup is disabled due to a bug");
+
     size_t decoderLayers = opt<size_t>("dec-depth");
     size_t decoderBaseDepth = opt<size_t>("dec-cell-base-depth");
     size_t decoderHighDepth = opt<size_t>("dec-cell-high-depth");
+    int decoderAttentionHops = opt<int>("dec-attention-hops");
+    ABORT_IF(decoderAttentionHops >= decoderBaseDepth,
+             "--dec-attention-hops must be lower than dec-cell-base-depth");
 
     // setting up conditional (transitional) cell
     auto baseCell = rnn::stacked_cell(graph);
-    for(size_t i = 1; i <= decoderBaseDepth; ++i) {
-      bool transition = (i > 2);
+
+    lastAttentionCellIds_ = std::vector<size_t>(state->getEncoderStates().size(), -1);
+    numBaseCells_ = 0;
+    for(int i = 1; i <= decoderBaseDepth; ++i) {
+      bool transition = (i > 1+decoderAttentionHops);
```

Comment:
This variable name is too broad, its meaning is not clear. Please choose a variable (and parameter) name that is more concise. Also comment precisely its meaning here.